### PR TITLE
Add dynamic prefix in configuration

### PR DESCRIPTION
**Pull request by ElevnLi**
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -72,7 +80,7 @@ services:
     command:
       - "--providers.docker.exposedByDefault=false"
       - "--providers.docker.network=cvat"
-      - "--entryPoints.web.address=:8080"
+      - "--entryPoints.web.address=:80"
```
Comment (jnothman) [inline docker-compose.yml:83]: Leave this as it was, and instead use a different port mapping below
```suggestion
      - "--entryPoints.web.address=:8080"
```

File: docker-compose.yml [hunk 2]
```diff
@@ -82,6 +90,7 @@ services:
     #   - traefik.http.routers.dashboard.service=api@internal
     #   - traefik.http.routers.dashboard.rule=Host(`${CVAT_HOST:-localhost}`)
     ports:
+      - 80:80
       - 8080:8080
```
Comment (jnothman) [inline docker-compose.yml:94]: [Store config in the environment](https://12factor.net/config)

```suggestion
      - ${WEB_PORT:-8080}:8080
```

**Review by jnothman — COMMENTED** [review]
Are you sure a lot of this could not be achieved with adding a [`basename`](https://v5.reactrouter.com/web/api/BrowserRouter/basename-string) prop to `<BrowserRouter>`?
That should automatically update all `<Link>`s. I assume that since `useHistory` is from the same package (`react-router`), it will also be aware of basename. What I'm not sure about is whether `antd.lib.button.Button` will/can use a react-router Link.

**Comment by jnothman** [comment]
Have you investigated whether this can be achieved with a [`basename`](https://v5.reactrouter.com/web/api/BrowserRouter/basename-string) prop to `<BrowserRouter>`?

**Comment by ElevnLi** [comment]
Better solution as #3

**Pull request closed without merging**